pkg/controller/deployment/deployment_controller.go

```diff
@@ -424,6 +424,27 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+// syncPausedDeployment allows scaling of the deployment (either done manually
+// or by autoscalers) to propagate to the active rc of the deployment.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, err := deploymentutil.GetNewRCFromList(deployment, dc.client,
+		func(namespace string, options api.ListOptions) ([]api.ReplicationController, error) {
+			return dc.rcStore.List()
+		})
+	if err != nil {
+		return err
+	}
+	if newRC == nil {
+		// There is no replication controller that matches the intent of
+		// this deployment which means that either the deployment has never
+		// run (created as paused), or it was switched to paused and has been
+		// modified since. Such is life.
+		return nil
+	}
+	_, err = dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
```

Comment:
fat fingers

> We don't want to scale the new RC to the full replica count of the Deployment. That would radically change the pace of the rolling update and violate maxSurge.

I was thinking about this today and I and believe maxSurge does make sense only under a progressing deployment, right? If a deployment is paused but not underway a rollout (ie. its latest rs owns all of the replicas), then we should scale to the full count. If a deployment is paused in the middle of a rollout, respect maxSurge/maxUnavailable and scale down older, scale up new. Also this is specific for rolling deployments.